cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1376,20 +1386,39 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
          */
         Future<ReconciliationState> zkScaleUpInSmallSteps() {
             return zkSetOperations.getAsync(namespace, ZookeeperCluster.zookeeperClusterName(name))
-                    .map(sts -> sts == null ? Integer.valueOf(0) : sts.getSpec().getReplicas())
-                    .compose(currentReplicas -> {
-                        if (currentReplicas > 0 && zkCluster.getReplicas() > currentReplicas) {
-                            zkCluster.setReplicas(currentReplicas + 1);
-                            zkScalingUp = true;
-                        }
+                    .compose(sts -> {
+                        if (sts != null) {
+                            int currentReplicas = sts.getSpec().getReplicas();
+                            if (currentReplicas > 0 && zkCluster.getReplicas() > currentReplicas) {
+                                // TODO: For ZK 3.5.x we need to limit scaling to 1 at a time. Should we enforce that here?
```

Comment:
Does this TODO still apply?

Reply:
Yes, someone could set the replica to two higher and both node would be added (without the roll). For stability we only want to add 1 node at time to the cluster. Which is still possible as things stand, but the procedure says only add one at a time.

Reply:
To prevent this (if the user specifies more than one extra node) I think we would need to overwrite the STS zookeeper spec and apply it. Do you think that is worth it or can we rely on the user following the procedure?

Reply:
I didn't realized these were connected at first. But main of this comment was whether we really need the TODO, especially with temp solution. You should either do it or remove the TODO.

Reply:
Ok, I have added logic at the start of the `zkScalingSetup` method to abort the reconciliation (with a status + log message) if a user tries to scale more than one zk server at a time.